File: Soving/BAEKJOON/bronze/B1_1259.py
# 팰린드롬수

while True:
    word = input()
    if word == '0':
        break
    answer = 'yes'
    for i in range(len(word)//2):
        if word[i]==word[-1-i]:
            continue
        answer = 'no'
        break
    if word[0] == '0':
        answer = 'no'

    print(answer)

File: java/backjoon/class/class2/B1_1259.java
import java.io.*;
import java.util.Objects;

public class B1_1259 {
    public static void main(String[] args) throws IOException {
        System.setIn(new FileInputStream("input.txt"));
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        while (true) {
            String S = sc.readLine();
            if (Objects.equals(S, "0")) {
                break;
            }
            System.out.println(solution(S));
        }
    }

    private static String solution(String S) {
        int start = 0;
        int end = S.length()-1;
        while (start<=end) {
            if (S.charAt(start) == S.charAt(end)) {
                start++;
                end--;
                continue;
            }
            return "no";
        }
        return "yes";
    }
}
